nano/node/node.cpp

```diff
@@ -2777,6 +2777,14 @@ void nano::active_transactions::confirm_frontiers (nano::transaction const & tra
 		size_t elections_count (0);
 		for (auto i (node.store.latest_begin (transaction_a, next_frontier_account)), n (node.store.latest_end ()); i != n && elections_count < max_elections; ++i)
 		{
+			{
+				std::lock_guard<std::mutex> guard (mutex);
+				if (stopped)
```

Comment:
May be better make stopped atomic, then it can be just `for (auto i (node.store.latest_begin (transaction_a, next_frontier_account)), n (node.store.latest_end ()); i != n && && !stopped && elections_count < max_elections; ++i)`

Reply:
I like that it is protected by a mutex otherwise it can still have timing issues and delays:

```
std::unique_lock<std::mutex> lk (mutex);
...
const auto extra_delay (std::min (roots.size (), max_broadcast_queue) * node.network.broadcast_interval_ms * 2);
if (stopped) // read stopped = false
{
  break;
}
// [another thread] `stopped` = true (no lock)
condition.wait_for (lock, std::chrono::milliseconds (node.network_params.network.request_interval_ms + extra_delay));
```

Reply:
Stopped still needs to be protected by a mutex in some scenarios otherwise you can still have delays, e.g:

```
std::unique_lock<std::mutex> lk (mutex);
...
const auto extra_delay (std::min (roots.size (), max_broadcast_queue) * node.network.broadcast_interval_ms * 2);
if (stopped) // read stopped = false
{
  break;
}
// [another thread] `stopped` = true (no lock)
// Still delay
condition.wait_for (lock, std::chrono::milliseconds (node.network_params.network.request_interval_ms + extra_delay));
``` 

However `stopped = true` currently is protected by a lock, so it should work as you've described and is what I've done.